bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeDeviceDescription.java

```diff
@@ -0,0 +1,535 @@
+    public static final String CHANNEL_TYPE_WINDOWPOSSENSOR = "Window Position Sensor";
+    public static final String CHANNEL_TYPE_TRIGGER = "Trigger";
+    public static final String CHANNEL_TYPE_DOOROPENERACTUATOR = "Door Opener Actuator";
+    public static final String CHANNEL_TYPE_RINGSENSOR = "Ring Sensor";
+    public static final String CHANNEL_TYPE_CORRIDORLIGHTSWITCH = "Corridor Light Switch";
+    public static final String CHANNEL_TYPE_MUTEACTUATOR = "Mute Actor";
+    public static final String CHANNEL_TYPE_DIMMINGACTUATOR = "Dimming Actuator";
+    public static final String CHANNEL_TYPE_SHUTTERACTUATOR = "Shutter Actuator";
+
+    // interface strings
+    public static final String DEVICE_INTERFACE_UNKNOWN_TYPE = "unknown";
+    public static final String DEVICE_INTERFACE_WIRELESS_TYPE = "wireless";
+    public static final String DEVICE_INTERFACE_VIRTUAL_TYPE = "virtual";
+    public static final String DEVICE_INTERFACE_WIRED_TYPE = "wired";
+    public static final String DEVICE_INTERFACE_HUE_TYPE = "hue";
+
+    public String deviceLabel = "";
+    // public String deviceTypeString = "";
+    public String deviceId = "";
+    public String interfaceType = "";
+    public boolean validDevice = false;
+
+    public List<FreeAtHomeDeviceChannel> listOfThings = new ArrayList<>();
+
+    public FreeAtHomeDeviceDescription() {
+        validDevice = false;
+    }
+
+    public FreeAtHomeDeviceDescription(JsonObject jsonObject, String id) {
+
+        // set the device ID
+        deviceId = id;
+
+        // set the device invalid at first
+        validDevice = false;
+
+        boolean sceneIsDetected = id.toLowerCase().startsWith("ffff48");
+        boolean ruleIsDetected = id.toLowerCase().startsWith("ffff4a");
+
+        JsonObject jsonObjectOfId = jsonObject.getAsJsonObject(id);
+
+        if (null == jsonObjectOfId) {
+            return;
+        }
+
+        JsonElement jsonObjectOfInterface = jsonObjectOfId.get("interface");
+
+        if (null != jsonObjectOfInterface) {
+            String interfaceString = jsonObjectOfInterface.getAsString();
+
+            if (interfaceString.toLowerCase().startsWith("vdev:")) {
+                interfaceType = DEVICE_INTERFACE_VIRTUAL_TYPE;
+            } else if (interfaceString.toLowerCase().startsWith("hue")) {
+                interfaceType = DEVICE_INTERFACE_HUE_TYPE;
+            } else if (interfaceString.toLowerCase().startsWith("rf")) {
+                interfaceType = DEVICE_INTERFACE_WIRELESS_TYPE;
+            } else if (interfaceString.toLowerCase().startsWith("tp")) {
+                interfaceType = DEVICE_INTERFACE_WIRED_TYPE;
+            } else {
+                interfaceType = DEVICE_INTERFACE_UNKNOWN_TYPE;
+            }
+        } else {
+            interfaceType = DEVICE_INTERFACE_UNKNOWN_TYPE;
+        }
+
+        JsonElement jsonObjectOfDeviceLabel = jsonObjectOfId.get("displayName");
+
+        if (null == jsonObjectOfDeviceLabel) {
```

Comment:
Why do you list null first? This seems pretty uncommon, especially as sometimes you do `constant ==` and sometimes `== constant`.

Reply:
solved... all "if" same. e.g. if ( var == true )